code/game/objects/effects/temporary_visuials/temproary_visual.dm

```diff
@@ -0,0 +1,39 @@
+//temporary visual effects
+/obj/effect/temp_visual
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "nothing"
+	anchored = TRUE
+	layer = ABOVE_MOB_LAYER
+	mouse_opacity = 0
+	var/duration = 10 //in deciseconds
+	var/randomdir = TRUE
+
+/obj/effect/temp_visual/initialize()
+	//. = ..()
+	//if(randomdir)
+	//	setDir(pick(GLOB.cardinals))       //tg stuff, readd when necessary
+
+	spawn(duration)
+		qdel(src)
+
+/obj/effect/temp_visual/singularity_act()
+	return
+
+/obj/effect/temp_visual/singularity_pull()
+	return
+
+/obj/effect/temp_visual/ex_act()
+	return
+
+/*
+/obj/effect/temp_visual/dir_setting
+	randomdir = FALSE
+
+/obj/effect/temp_visual/dir_setting/Initialize(mapload, set_dir)
+	if(set_dir)
+		setDir(set_dir)
+	. = ..()
+*/		//More tg stuff that might be useful later
+
+/obj/effect/temp_visual/proc/setDir(var/newdir)
```

Comment:
We already have `/atom/proc/set_dir(new_dir)`  please remove this and use that instead.